src/lsptoolshost/projectRestore/restore.ts

```diff
@@ -18,31 +18,52 @@ import { getCSharpDevKit } from '../../utils/getCSharpDevKit';
 
 let _restoreInProgress = false;
 
-export function registerRestoreCommands(context: vscode.ExtensionContext, languageServer: RoslynLanguageServer) {
-    if (getCSharpDevKit()) {
-        // We do not need to register restore commands if using C# devkit.
-        return;
+export function registerRestoreCommands(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    csharpOutputChannel: vscode.LogOutputChannel
+) {
+    // We do not need to register restore commands if using C# devkit.
+    if (!getCSharpDevKit()) {
+        context.subscriptions.push(
+            vscode.commands.registerCommand('dotnet.restore.project', async (_request): Promise<void> => {
+                return chooseProjectAndRestore(languageServer, csharpOutputChannel);
+            })
+        );
+        context.subscriptions.push(
+            vscode.commands.registerCommand('dotnet.restore.all', async (): Promise<void> => {
+                return restore(languageServer, csharpOutputChannel, [], true);
+            })
+        );
     }
-    const restoreChannel = vscode.window.createOutputChannel(vscode.l10n.t('.NET NuGet Restore'));
-    context.subscriptions.push(
-        vscode.commands.registerCommand('dotnet.restore.project', async (_request): Promise<void> => {
-            return chooseProjectAndRestore(languageServer, restoreChannel);
-        })
-    );
-    context.subscriptions.push(
-        vscode.commands.registerCommand('dotnet.restore.all', async (): Promise<void> => {
-            return restore(languageServer, restoreChannel, [], true);
-        })
-    );
 
     languageServer.registerOnRequest(ProjectNeedsRestoreRequest.type, async (params) => {
-        await restore(languageServer, restoreChannel, params.projectFilePaths, false);
+        let projectFilePaths = params.projectFilePaths;
+        if (getCSharpDevKit()) {
+            // Only restore '.cs' files (file-based apps) if CDK is loaded.
```

Comment:
Why do we need to have this check at all? Is there a case we'd be sending over a request for a restore if C# DevKit was the one loading projects?